lib/runner/variant-reassignment.js

```diff
@@ -98,7 +101,18 @@ export default class VariantReassignment {
       await this.productService.createProduct(anonymizedProductDraft)
     }
     // e.g. Example 7
-    await this._ensureSlugUniqueness(productDraft, matchingProducts)
+    await this._ensureSlugUniqueness(productDraft,
+      matchingProducts.filter(product => product.id !== ctpProductToUpdate.id))
+  }
+
+  _replaceProductInProductArray (productToReplace, productArray) {
+    const productToReplaceSkus = this.productService.getProductSkus(productToReplace)
+    return productArray.map((product) => {
+      const productSkus = this.productService.getProductSkus(product)
```

Comment:
can we use `id` to compare if the products are equal instead of looping through variants in every product? Also, can we use `for-in` loop which will break once it finds the matching product so we don't loop through all products if the match is in the first index?

Reply:
We can't use id because after productType changes, the productArray will contain product with different ID than  `productToReplace`.

Reply:
This method is now removed completely as we don't need it anymore.

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/commits/0c7949f77601e71500de9d783a18cd65fb1e4f0d